### ISSUE TITLE
`java.lang.StringIndexOutOfBoundsException` in various files

### ISSUE THREAD

**Issue by ckipp01** [scalameta/scalafmt#2037]
This template is a guideline, not a strict requirement.
- **Version**: 0.6.0
- **Integration**: CLI and Metals
- **Configuration**:
```
version = "2.6.0"
project.git = true
align.preset = none
align.stripMargin = true
docstrings = Asterisk
assumeStandardLibraryStripMargin = true
binPack.literalsExclude = ["Term.Name"]
onTestFailure = """
To fix this problem:
1. run ./bin/scalafmt from the project root directory
2. `git add . && git commit --amend` OR `git commit -am "Run scalafmt"`
"""
project.excludeFilters = [
  "test-workspace"
  "metals-bench/src/main/resources"
  "mtags/src/main/scala-3/scala/meta/internal/mtags/MtagsEnrichments.scala"
  "tests/unit/src/test/resources"
]
```

## Steps

The easiest way to see this is to probably run it on the Metals codebase. I apologize, I haven't been able to nail down exactly what is causing this, but when we updated to 2.6.0 in Metals, we are seeing a few `java.lang.StringIndexOutOfBoundsException`s in various files. You can see this happening [here in the CI](https://github.com/scalameta/metals/runs/787594121?check_suite_focus=true#step:6:27). It looks like the following files are causing the issues:

- https://github.com/scalameta/metals/blob/master/tests/cross/src/test/scala/tests/pc/SignatureHelpDocSuite.scala
- https://github.com/scalameta/metals/blob/master/tests/slow/src/test/scala/tests/gradle/GradleLspSuite.scala
- https://github.com/scalameta/metals/blob/master/tests/slow/src/test/scala/tests/sbt/SbtLspSuite.scala

The exception for all of the files seems to be the same:

```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at scala.collection.StringOps$.apply$extension(StringOps.scala:170)
	at org.scalafmt.internal.FormatWriter$FormatLocations$Entry.$anonfun$formatMarginized$4(FormatWriter.scala:367)
	at org.scalafmt.internal.FormatWriter$FormatLocations$Entry.$anonfun$formatMarginized$4$adapted(FormatWriter.scala:360)
	at scala.Option.map(Option.scala:242)
	at org.scalafmt.internal.FormatWriter$FormatLocations$Entry.$anonfun$formatMarginized$3(FormatWriter.scala:360)
	at scala.Option.flatMap(Option.scala:283)
	at org.scalafmt.internal.FormatWriter$FormatLocations$Entry.formatMarginized(FormatWriter.scala:359)
	at org.scalafmt.internal.FormatWriter.$anonfun$mkString$1(FormatWriter.scala:59)
	at org.scalafmt.internal.FormatWriter.$anonfun$mkString$1$adapted(FormatWriter.scala:45)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1279)
	at org.scalafmt.internal.FormatWriter.mkString(FormatWriter.scala:45)
	at org.scalafmt.Scalafmt$.formatCode(Scalafmt.scala:77)
	at org.scalafmt.cli.ScalafmtCoreRunner$.unsafeHandleFile(ScalafmtCoreRunner.scala:91)
	at org.scalafmt.cli.ScalafmtCoreRunner$.handleFile(ScalafmtCoreRunner.scala:70)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5(ScalafmtCoreRunner.scala:46)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5$adapted(ScalafmtCoreRunner.scala:45)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1279)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:946)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:52)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:97)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:55)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:49)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:943)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal(Tasks.scala:169)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.internal$(Tasks.scala:156)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.internal(Tasks.scala:303)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:149)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:148)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:303)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

## Problem

This doesn't break anything, but more of an fyi since these files aren't being formatted because of the exception.

## Expectation

No exception thrown and file files formatted beautifully.

**Comment by kitbellew** [comment]
fixed in #2029

**Comment by ckipp01** [comment]
Ah I didn’t see that. Thanks for the heads up!

**Comment by kitbellew** [comment]
> Ah I didn’t see that. Thanks for the heads up!

thank you for reporting, and of course you are not expected to have seen that :) released in 2.6.1, please check.
